metis-indexing/src/main/java/eu/europeana/indexing/utils/SetUtils.java

```diff
@@ -19,12 +17,48 @@ private SetUtils() {}
 
   /**
    * <p>
-   * This method generates all possible combinations of elements of different sets. Each combination
-   * consists of zero or one element of each set. As such, the number of combinations of sets s1,
-   * s2, ..., sn that will be generated is equal to (|s1| + 1) * (|s2| + 1) * ... * (|sn| + 1),
-   * where |s| denotes the cardinality (size) of set s. However, depending on the choice of the
-   * types and the concatenator, not all of these may be unique and the result set may contain fewer
-   * items.
+   * This method generates all possible combinations of elements of different sets.
+   * </p>
+   * <p>
+   * Each combination consists of zero or one element of each set. As such, the number of
+   * combinations of sets s1, s2, ..., sn that will be generated is equal to (|s1| + 1) * (|s2| + 1)
+   * * ... * (|sn| + 1), where |s| denotes the cardinality (size) of set s. However, depending on
+   * the choice of the types and the concatenator, not all of these may be unique and the result set
+   * may contain fewer items.
+   * </p>
+   * <p>
+   * It follows that if one of the input sets is empty, the result will be as if that set was not
```

Comment:
I know this is probably not happening but perhaps we could check if the list contains an empty set early so that we avoid computations that are not required..